File: backend/courts/application/use_cases/update_court.py
from typing import Dict, Any, Optional, List
from ...domain.repositories.court_repository import ICourtRepository
from ...models import Court # Asumiendo que Court está en backend/courts/models.py

class UpdateCourtUseCase:
    """
    Caso de uso para actualizar una cancha existente.
    Esta clase reside en la capa de Aplicación.
    """
    def __init__(self, court_repository: ICourtRepository):
        self.court_repository = court_repository

    async def execute(self, court_id: int, court_data: Dict[str, Any], images_data: Optional[List[Any]] = None, images_to_delete: Optional[List[int]] = None) -> Optional[Court]:
        """
        Ejecuta el caso de uso para actualizar una cancha.

        Args:
            court_id (int): El ID de la cancha a actualizar.
            court_data (Dict[str, Any]): Los datos de la cancha a actualizar.
            images_data (Optional[List[Any]]): Lista de nuevos archivos de imagen a añadir.
            images_to_delete (Optional[List[int]]): Lista de IDs de imágenes existentes a eliminar.

        Returns:
            Optional[Court]: La cancha actualizada o None si no se encontró.
        """
        # Aquí se podría añadir lógica de aplicación adicional si fuera necesario.
        return await self.court_repository.update(court_id, court_data, images_data, images_to_delete)

File: frontend/src/application/use-cases/courts/update-court.js
import { ICourtRepository } from '../../../domain/repositories/court-repository'; // Importar la interfaz del repositorio

/**
 * Caso de uso para actualizar una cancha existente.
 * Esta clase reside en la capa de Aplicación y orquesta el proceso de actualización de cancha
 * utilizando la interfaz del repositorio de Dominio.
 */
export class UpdateCourtUseCase {
  /**
   * @param {ICourtRepository} courtRepository - Una implementación del repositorio de canchas.
   */
  constructor(courtRepository) {
    if (!(courtRepository instanceof ICourtRepository)) {
      throw new Error('courtRepository must be an instance of ICourtRepository');
    }
    this.courtRepository = courtRepository;
  }

  /**
   * Ejecuta el caso de uso para actualizar una cancha.
   * @param {number} courtId - El ID de la cancha a actualizar.
   * @param {object} courtData - Los datos para actualizar la cancha.
   * @returns {Promise<object>} Una promesa que resuelve con los datos de la cancha actualizada.
   */
  async execute(courtId, courtData) {
    // Aquí se podría añadir lógica de aplicación adicional si fuera necesario
    // (ej. validaciones, notificaciones, etc.) antes o después de llamar al repositorio.
    return this.courtRepository.updateCourt(courtId, courtData);
  }
}
